### PR TITLE
slideshow: fetch correct layer for AnimatedElement

### PR DESCRIPTION
**Pull request by eszkadev**
- fix bug: some animations were not added to the queue
- provide layer to the AnimetedElement

Core side patches:
- https://gerrit.libreoffice.org/c/core/+/173050    (initVisibility property for hiding objects which will appear later)

### CODE REVIEW

File: browser/src/slideshow/engine/SlideAnimations.ts [hunk 1]
```diff
@@ -40,14 +40,16 @@ function getNodeChildren(aNode: ContainerNodeInfo): Array<AnimationNodeInfo> {
 function createAnimationTree(
 	aAnimationRoot: AnimationNodeInfo,
 	aNodeContext: NodeContext,
+	metaSlide: MetaSlide,
 ): BaseNode {
-	return createAnimationNode(aAnimationRoot, null, aNodeContext);
+	return createAnimationNode(aAnimationRoot, null, aNodeContext, metaSlide);
```
Comment (mcecchetti) [inline browser/src/slideshow/engine/SlideAnimations.ts:45]: I think that would be cleaner, to pass the meta slide or just slide info to SlideAnimation ctor and append it to node context which is already passed down along tree nodes.
Anyway nice job.

File: browser/src/slideshow/engine/animationnodes/BaseNode.ts [hunk 2]
```diff
@@ -425,7 +425,6 @@ abstract class BaseNode {
 
 	public init(): boolean {
 		this.DBG(this.callInfo('init'));
-		if (!this.checkValidNode()) return false;
```
Comment (mcecchetti) [inline browser/src/slideshow/engine/animationnodes/BaseNode.ts:428]: To be double check. It can be dangerous.

Comment (eszkadev) [reply browser/src/slideshow/engine/animationnodes/BaseNode.ts:428]: I added some comment in the commit: https://github.com/CollaboraOnline/online/pull/10009/commits/55ad9ae47ac84d43924b91bf2fa106031ebbde64

do you know maybe what else could be missing instead? it was not allowing to resolve root node in some cases and we were missing animations on click in the slideshow

File: browser/src/slideshow/engine/SlideAnimations.ts (hunk 1 above)
Comment (eszkadev) [reply browser/src/slideshow/engine/SlideAnimations.ts]: right, done

File: browser/src/slideshow/engine/animationnodes/BaseNode.ts (hunk 2 above)
Comment (eszkadev) [reply browser/src/slideshow/engine/animationnodes/BaseNode.ts:428]: the root problem might be caused by:
```
DBG: AnimationTransformNode( 6, Invalid ).init
global.js:545 createPropertyAnimation: attribute is unknown: transform
global.js:545 AnimationTransformNode.createActivity: failed to create animation.
```
so then we are in the invalid state...

Comment (eszkadev) [reply browser/src/slideshow/engine/animationnodes/BaseNode.ts]: added proper solution now

Comment (mcecchetti) [reply browser/src/slideshow/engine/animationnodes/BaseNode.ts]: Thanks, the problem was caused by capitalized transform name I guess.

**Review by mcecchetti — APPROVED** [review]
Now looks good to me.

**Pull request merged**